e2e_tests/e2e_test.go

```diff
@@ -0,0 +1,299 @@
+		fmt.Printf("error reported during helm install: %v\n", err)
+		retryUpgrade := action.NewUpgrade(actionConfig)
+		retryUpgrade.Namespace = "default"
+		retryUpgrade.Install = true
+		require.Eventually(t, func() bool {
+			_, err = retryUpgrade.Run("sock", chart, values)
+			if err != nil {
+				fmt.Printf("error reported during helm upgrade: %v\n", err)
+			}
+			return err == nil
+		}, 3*time.Minute, 30*time.Second)
+	}
+
+	waitForAllDeploymentsToStart(t, clientset)
+
+	deployments := clientset.AppsV1().Deployments("default")
+
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	stream, err := os.ReadFile(filepath.Join("testdata", "nodejs", "deployment.yaml"))
+	require.NoError(t, err)
+	deployment, _, err := decode(stream, nil, nil)
+	require.NoError(t, err)
+	_, err = deployments.Create(context.Background(), deployment.(*appsv1.Deployment), metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	waitForAllDeploymentsToStart(t, clientset)
+
+	tracesConsumer := new(consumertest.TracesSink)
+	metricsConsumer := new(consumertest.MetricsSink)
+	logsConsumer := new(consumertest.LogsSink)
+	wantEntries := 3 // Minimal number of traces, metrics, and logs to wait for.
+	waitForData(t, wantEntries, tracesConsumer, metricsConsumer, logsConsumer)
+
+	replaceWithStar := func(string) string { return "*" }
+	shortenNames := func(value string) string {
+		if strings.HasPrefix(value, "kube-proxy") {
+			return "kube-proxy"
+		}
+		if strings.HasPrefix(value, "local-path-provisioner") {
+			return "local-path-provisioner"
+		}
+		if strings.HasPrefix(value, "kindnet") {
+			return "kindnet"
+		}
+		if strings.HasPrefix(value, "coredns") {
+			return "coredns"
+		}
+		if strings.HasPrefix(value, "otelcol") {
+			return "otelcol"
+		}
+		if strings.HasPrefix(value, "sock-splunk-otel-collector-agent") {
+			return "sock-splunk-otel-collector-agent"
+		}
+		if strings.HasPrefix(value, "sock-splunk-otel-collector-k8s-cluster-receiver") {
+			return "sock-splunk-otel-collector-k8s-cluster-receiver"
+		}
+		if strings.HasPrefix(value, "cert-manager-cainjector") {
+			return "cert-manager-cainjector"
+		}
+		if strings.HasPrefix(value, "sock-operator") {
+			return "sock-operator"
+		}
+		if strings.HasPrefix(value, "nodejs-test") {
+			return "nodejs-test"
+		}
+		if strings.HasPrefix(value, "cert-manager-webhook") {
+			return "cert-manager-webhook"
+		}
+		if strings.HasPrefix(value, "cert-manager") {
+			return "cert-manager"
+		}
+
+		return value
+	}
+	containerImageShorten := func(value string) string {
+		return value[(strings.LastIndex(value, "/") + 1):]
+	}
+
+	latestTrace := tracesConsumer.AllTraces()[len(tracesConsumer.AllTraces())-1]
+	actualSpan := latestTrace.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0)
+	host, _ := actualSpan.Attributes().Get("http.host")
+	assert.Equal(t, "localhost:3000", host.AsString())
+
+	statusCode, _ := actualSpan.Attributes().Get("http.status_code")
+	assert.Equal(t, "200", statusCode.AsString())
```

Comment:
Why checking this additionally? I see the expected files have these attributes

Reply:
I removed this entirely in the next version. Good point.